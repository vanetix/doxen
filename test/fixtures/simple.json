[
  {
    "tags": [],
    "description": {
      "full": "<p>Module dependencies</p>",
      "summary": "<p>Module dependencies</p>",
      "body": ""
    },
    "ignore": false,
    "code": "var fs = require('fs'),\n    crypto = require('crypto'),\n    Handlebars = require('handlebars');",
    "ctx": {
      "type": "declaration",
      "name": "fs",
      "value": "require('fs'),",
      "string": "fs"
    }
  },
  {
    "tags": [],
    "description": {
      "full": "<p>Global templates</p>",
      "summary": "<p>Global templates</p>",
      "body": ""
    },
    "ignore": false,
    "code": "var T = {\n  link: Handlebars.compile(fs.readFileSync(\n                __dirname + '/templates/link.html', 'utf8')),\n  layout: Handlebars.compile(fs.readFileSync(\n                __dirname + '/templates/layout.html', 'utf8')),\n  section: Handlebars.compile(fs.readFileSync(\n                __dirname + '/templates/section.html', 'utf8'))\n};",
    "ctx": {
      "type": "declaration",
      "name": "T",
      "value": "{",
      "string": "T"
    }
  },
  {
    "tags": [
      {
        "type": "param",
        "types": [
          "String",
          "Object"
        ],
        "name": "dox",
        "description": ""
      },
      {
        "type": "return",
        "types": [
          "String"
        ],
        "description": ""
      }
    ],
    "description": {
      "full": "<p>Expose <code>Doxen</code><br />- Returns an html representation of dox object <code>dox</code></p>",
      "summary": "<p>Expose <code>Doxen</code><br />- Returns an html representation of dox object <code>dox</code></p>",
      "body": ""
    },
    "isPrivate": false,
    "ignore": false,
    "code": "var Doxen = module.exports = function doxen(dox) {\n  if(typeof dox !== 'object') {\n    try {\n      dox = JSON.parse(json);\n    } catch(e) {\n      return 'Invalid json';\n    }\n  }\n\n  return build(dox);\n};",
    "ctx": {
      "type": "declaration",
      "name": "Doxen",
      "value": "module.exports = function doxen(dox) {",
      "string": "Doxen"
    }
  },
  {
    "tags": [
      {
        "type": "param",
        "types": [
          "Object"
        ],
        "name": "dox",
        "description": ""
      },
      {
        "type": "param",
        "types": [
          "Object"
        ],
        "name": "data",
        "description": ""
      }
    ],
    "description": {
      "full": "<p>Expose the full render functionality<br />- Rendered the given <code>obj</code> to the template</p>",
      "summary": "<p>Expose the full render functionality<br />- Rendered the given <code>obj</code> to the template</p>",
      "body": ""
    },
    "isPrivate": false,
    "ignore": false,
    "code": "Doxen.render = function render(dox, data) {\n  data = data || {};\n  data.title = data.title || 'Doxen';\n  data.content = data.content || build(dox);\n\n  return T.layout(data);\n}",
    "ctx": {
      "type": "method",
      "receiver": "Doxen",
      "name": "render",
      "string": "Doxen.render()"
    }
  },
  {
    "tags": [
      {
        "type": "param",
        "types": [
          "Object"
        ],
        "name": "dox",
        "description": ""
      },
      {
        "type": "return",
        "types": [
          "Object"
        ],
        "description": ""
      }
    ],
    "description": {
      "full": "<p>Build docs</p>",
      "summary": "<p>Build docs</p>",
      "body": ""
    },
    "isPrivate": false,
    "ignore": false,
    "code": "function build(dox) {\n  var i, len,\n      nav = '<nav><dl>',\n      docs = '<div>';\n\n  for(i = 0, len = dox.length; i < len; i = i + 1) {\n    nav += navigation(dox[i]);\n    docs += block(dox[i]);\n  }\n\n  nav += '</dl></nav>';\n  docs += '</div>';\n\n  return nav + docs;\n}",
    "ctx": {
      "type": "function",
      "name": "build",
      "string": "build()"
    }
  },
  {
    "tags": [
      {
        "type": "param",
        "types": [
          "Object"
        ],
        "name": "chunk",
        "description": ""
      },
      {
        "type": "return",
        "types": [
          "String"
        ],
        "description": ""
      }
    ],
    "description": {
      "full": "<p>Build the navigation pane for a block</p>",
      "summary": "<p>Build the navigation pane for a block</p>",
      "body": ""
    },
    "isPrivate": false,
    "ignore": false,
    "code": "function navigation(chunk) {\n  var data = {\n    id: makeId(chunk.code),\n    title: chunk.ctx.string,\n    description: chunk.description.summary\n  };\n\n  return T.link(data);\n}",
    "ctx": {
      "type": "function",
      "name": "navigation",
      "string": "navigation()"
    }
  },
  {
    "tags": [
      {
        "type": "param",
        "types": [
          "Object"
        ],
        "name": "chunk",
        "description": ""
      },
      {
        "type": "return",
        "types": [
          "String"
        ],
        "description": ""
      }
    ],
    "description": {
      "full": "<p>Build the section for given <code>chunk</code></p>",
      "summary": "<p>Build the section for given <code>chunk</code></p>",
      "body": ""
    },
    "isPrivate": false,
    "ignore": false,
    "code": "function block(chunk) {\n  var data = {\n    code: chunk.code,\n    title: chunk.ctx.string,\n    id: makeId(chunk.code),\n    type: chunk.ctx.type,\n    description: chunk.description.full,\n    params: chunk.tags.filter(function(tag) {\n      return tag.type === 'param';\n    }).map(function(tag) {\n      return {\n        types: tag.types.join(', '),\n        name: tag.name,\n        description: tag.description\n      };\n    })\n  };\n  \n  return T.section(data);\n}",
    "ctx": {
      "type": "function",
      "name": "block",
      "string": "block()"
    }
  },
  {
    "tags": [
      {
        "type": "param",
        "types": [
          "String"
        ],
        "name": "code",
        "description": ""
      },
      {
        "type": "return",
        "types": [
          "String"
        ],
        "description": ""
      }
    ],
    "description": {
      "full": "<p>Return a hash of <code>code</code> to use for element id</p>",
      "summary": "<p>Return a hash of <code>code</code> to use for element id</p>",
      "body": ""
    },
    "isPrivate": false,
    "ignore": false,
    "code": "function makeId(code) {\n  return crypto.createHash('md5').update(code).digest('hex').slice(0, 10);\n}",
    "ctx": {
      "type": "function",
      "name": "makeId",
      "string": "makeId()"
    }
  }
]