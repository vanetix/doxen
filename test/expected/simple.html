<dl><dt><a href="#657d878df3">fs</a></dt>
<dd>&lt;p&gt;Module dependencies&lt;/p&gt;</dd>
<dt><a href="#fa4b04a163">T</a></dt>
<dd>&lt;p&gt;Global templates&lt;/p&gt;</dd>
<dt><a href="#8c8b3c828e">Doxen</a></dt>
<dd>&lt;p&gt;Expose &lt;code&gt;Doxen&lt;/code&gt;&lt;br /&gt;- Returns an html representation of dox object &lt;code&gt;dox&lt;/code&gt;&lt;/p&gt;</dd>
<dt><a href="#105e6d85f5">Doxen.render()</a></dt>
<dd>&lt;p&gt;Expose the full render functionality&lt;br /&gt;- Rendered the given &lt;code&gt;obj&lt;/code&gt; to the template&lt;/p&gt;</dd>
<dt><a href="#a52a606ff6">build()</a></dt>
<dd>&lt;p&gt;Build docs&lt;/p&gt;</dd>
<dt><a href="#ae07325fdb">navigation()</a></dt>
<dd>&lt;p&gt;Build the navigation pane for a block&lt;/p&gt;</dd>
<dt><a href="#5492d808f6">block()</a></dt>
<dd>&lt;p&gt;Build the section for given &lt;code&gt;chunk&lt;/code&gt;&lt;/p&gt;</dd>
<dt><a href="#8cacfc7c15">makeId()</a></dt>
<dd>&lt;p&gt;Return a hash of &lt;code&gt;code&lt;/code&gt; to use for element id&lt;/p&gt;</dd>
</dl><div><section id="657d878df3">
  <h1>fs</h1>
  <h2>declaration</h2>

  

  
    <p>Module dependencies</p>
  

  <pre>
    <code>var fs = require('fs'),
    crypto = require('crypto'),
    Handlebars = require('handlebars');</code>
  </pre>
</section>
<section id="fa4b04a163">
  <h1>T</h1>
  <h2>declaration</h2>

  

  
    <p>Global templates</p>
  

  <pre>
    <code>var T = {
  link: Handlebars.compile(fs.readFileSync(
                __dirname + '/templates/link.html', 'utf8')),
  layout: Handlebars.compile(fs.readFileSync(
                __dirname + '/templates/layout.html', 'utf8')),
  section: Handlebars.compile(fs.readFileSync(
                __dirname + '/templates/section.html', 'utf8'))
};</code>
  </pre>
</section>
<section id="8c8b3c828e">
  <h1>Doxen</h1>
  <h2>declaration</h2>

  
    <dl>
    
      <dt>dox</dt>
      <dd></dd>
    
    </dl>
  

  
    <p>Expose <code>Doxen</code><br />- Returns an html representation of dox object <code>dox</code></p>
  

  <pre>
    <code>var Doxen = module.exports = function doxen(dox) {
  if(typeof dox !== 'object') {
    try {
      dox = JSON.parse(json);
    } catch(e) {
      return 'Invalid json';
    }
  }

  return build(dox);
};</code>
  </pre>
</section>
<section id="105e6d85f5">
  <h1>Doxen.render()</h1>
  <h2>method</h2>

  
    <dl>
    
      <dt>dox</dt>
      <dd></dd>
    
      <dt>data</dt>
      <dd></dd>
    
    </dl>
  

  
    <p>Expose the full render functionality<br />- Rendered the given <code>obj</code> to the template</p>
  

  <pre>
    <code>Doxen.render = function render(dox, data) {
  data = data || {};
  data.title = data.title || 'Doxen';
  data.content = data.content || build(dox);

  return T.layout(data);
}</code>
  </pre>
</section>
<section id="a52a606ff6">
  <h1>build()</h1>
  <h2>function</h2>

  
    <dl>
    
      <dt>dox</dt>
      <dd></dd>
    
    </dl>
  

  
    <p>Build docs</p>
  

  <pre>
    <code>function build(dox) {
  var i, len,
      nav = '<dl>',
      docs = '<div>';

  for(i = 0, len = dox.length; i < len; i = i + 1) {
    nav += navigation(dox[i]);
    docs += block(dox[i]);
  }

  nav += '</dl>';
  docs += '</div>';

  return nav + docs;
}</code>
  </pre>
</section>
<section id="ae07325fdb">
  <h1>navigation()</h1>
  <h2>function</h2>

  
    <dl>
    
      <dt>chunk</dt>
      <dd></dd>
    
    </dl>
  

  
    <p>Build the navigation pane for a block</p>
  

  <pre>
    <code>function navigation(chunk) {
  var data = {
    id: makeId(chunk.code),
    title: chunk.ctx.string,
    description: chunk.description.short
  };

  return T.link(data);
}</code>
  </pre>
</section>
<section id="5492d808f6">
  <h1>block()</h1>
  <h2>function</h2>

  
    <dl>
    
      <dt>chunk</dt>
      <dd></dd>
    
    </dl>
  

  
    <p>Build the section for given <code>chunk</code></p>
  

  <pre>
    <code>function block(chunk) {
  var data = {
    code: chunk.code,
    title: chunk.ctx.string,
    id: makeId(chunk.code),
    type: chunk.ctx.type,
    description: chunk.description.full,
    params: chunk.tags.filter(function(tag) {
      return tag.type === 'param';
    }).map(function(tag) {
      return {
        types: tag.types.join(', '),
        name: tag.name,
        description: tag.description
      };
    })
  };
  
  return T.section(data);
}</code>
  </pre>
</section>
<section id="8cacfc7c15">
  <h1>makeId()</h1>
  <h2>function</h2>

  
    <dl>
    
      <dt>code</dt>
      <dd></dd>
    
    </dl>
  

  
    <p>Return a hash of <code>code</code> to use for element id</p>
  

  <pre>
    <code>function makeId(code) {
  return crypto.createHash('md5').update(code).digest('hex').slice(0, 10);
}</code>
  </pre>
</section>
</div>